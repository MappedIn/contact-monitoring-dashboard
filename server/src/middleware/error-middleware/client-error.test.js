const ClientError = require('./client-error');

describe('client-error', () => {
	test('has expected types enum', () => {
		expect(ClientError.Types).toMatchSnapshot();
	});

	test('client error has default bad request code and is type Error', () => {
		const clientError = new ClientError();
		expect(clientError instanceof Error).toBe(true);
		expect(clientError.code).toBe(ClientError.Types.BAD_REQUEST);
	});

	test('can override error message and code', () => {
		const clientError = new ClientError({
			code: 100,
			message: 'test error',
		});

		expect(clientError.code).toBe(100);
		expect(clientError.message).toBe('test error');
	});
});
