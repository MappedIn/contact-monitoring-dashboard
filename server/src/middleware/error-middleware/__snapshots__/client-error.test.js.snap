// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client-error has expected types enum 1`] = `
Object {
  "BAD_REQUEST": 400,
  "CONFLICT": 409,
  "FORBIDDEN": 403,
  "METHOD_NOT_ALLOWED": 405,
  "NOT_ACCEPTABLE": 406,
  "NOT_FOUND": 404,
  "UNAUTHORIZED": 401,
  "UNPROCESSABLE_ENTITY": 422,
}
`;
